### PR TITLE
Teclis

### PR DESCRIPTION
**Pull request by tmagicturtle**
I'm not entirely sure what [Rewind] in his thread title meant. If it _didn't_ mean that he didn't want to use it anymore, then this pull request should be good. I updated his Pokemon to Darkrai, added the Z-Crystal, Z-Move, and terrain effect.

### CODE REVIEW

File: data/mods/ssb/random-teams.js [hunk 1]
```diff
@@ -535,10 +535,10 @@ class RandomStaffBrosTeams extends RandomTeams {
 				evs: {spa: 252, spd: 4, spe: 252}, ivs: {atk: 0}, nature: 'Timid',
 			},
 			'Teclis': {
-				species: 'Gallade', ability: 'Intimidate', item: 'Galladite', gender: 'M',
-				moves: ['Sacred Sword', 'Psycho Cut', 'Knock Off'],
-				signatureMove: 'Zekken',
-				evs: {atk: 252, def: 4, spe: 252}, nature: 'Jolly', shiny: true,
+				species: 'Darkrai', ability: 'Dark Aura', item: 'Darkrainium Z', gender: 'M',
```
Comment (HoeenCoder) [inline data/mods/ssb/random-teams.js]: Genderless (use `'N'`)

File: data/mods/ssb/moves.js [hunk 2]
```diff
@@ -3306,31 +3306,85 @@ let BattleMovedex = {
...
+		terrain: 'nightmarefield',
+		effect: {
+			duration: 5,
+			durationCallback(source, effect) {
+				if (source && source.hasItem('terrainextender')) {
+					return 8;
+				}
+				return 5;
+			},
+			onTryMove(target, source, move) {
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Empty block?

File: data/mods/ssb/moves.js [hunk 3]
```diff
@@ -3306,31 +3306,85 @@ let BattleMovedex = {
...
+		shortDesc: "5 turns. Sleeping Pokemon get Nightmare.",
+		id: "nightmarefield",
+		name: "Nightmare Field",
+		isNonstandard: "Custom",
+		pp: 10,
+		priority: 0,
+		flags: {},
+		terrain: 'nightmarefield',
+		effect: {
+			duration: 5,
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Supposed to last 4 turns, dont support terrain extender as a result.

File: data/mods/ssb/moves.js [hunk 4]
```diff
@@ -3306,31 +3306,85 @@ let BattleMovedex = {
...
-		boosts: {
-			atk: 2,
-			spe: 1,
+		status: 'slp',
+		isZ: "darkrainiumz",
+		secondary: {
+			chance: 100,
+			self: {
+				onHit() {
+					this.field.setTerrain('nightmarefield');
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Id run this in an onHit rather than as a secondary, prevents possible suppression of secondaries.

**Review by HoeenCoder — COMMENTED** [review]
Again, this is tested right?

File: data/mods/ssb/random-teams.js (hunk 1 above)
Comment (tmagicturtle) [reply data/mods/ssb/random-teams.js]: Whoops. Thank you

File: data/mods/ssb/moves.js [hunk 5]
```diff
@@ -3352,16 +3347,7 @@ let BattleMovedex = {
 		flags: {},
 		terrain: 'nightmarefield',
```
Comment (ghost) [inline data/mods/ssb/moves.js]: pseudoWeather over terrain imo

Comment (ghost) [inline data/mods/ssb/moves.js]: pseudoWeather over terrain imo

Comment (tmagicturtle) [reply data/mods/ssb/moves.js]: Done.

Comment (tmagicturtle) [reply data/mods/ssb/moves.js]: Wait. Do not merge. Nightmare effect not activating.

Comment (ghost) [reply data/mods/ssb/moves.js]: My b, add this to nightmare configuration

```suggestion
		onAfterMoveSecondarySelf: function () {
		this.addPseudoWeather('nightmarefield');
		},
```

File: data/mods/ssb/moves.js [hunk 6]
```diff
@@ -3337,15 +3335,14 @@ let BattleMovedex = {
...
-		shortDesc: "5 turns. Sleeping Pokemon get Nightmare.",
+		desc: "For 4 turns, Nightmare Field is active. During the effect, sleeping Pokemon suffer from the effects of Nightmare.",
+		shortDesc: "4 turns. Sleeping Pokemon get Nightmare.",
 		id: "nightmarefield",
 		name: "Nightmare Field",
 		isNonstandard: "Custom",
 		pp: 10,
 		priority: 0,
 		flags: {},
-		terrain: 'nightmarefield',
```
Comment (ghost) [inline data/mods/ssb/moves.js]: Also here, add pseudoWeather instead of terrain

Comment (ghost) [reply data/mods/ssb/moves.js]: You do need that or else it wont work,

File: data/mods/ssb/moves.js (hunk 5 above)
Comment (ghost) [reply data/mods/ssb/moves.js]: Make sure its properly tabbed ofc

File: data/mods/ssb/moves.js [hunk 7]
```diff
@@ -3370,8 +3371,7 @@ let BattleMovedex = {
 		secondary: null,
 		target: "self",
 		type: "Dark",
-	},
-	// tennisace
+	},	// tennisace
```
Comment (ghost) [inline data/mods/ssb/moves.js]:
```suggestion
	},
	// tennisace
```

Comment (tmagicturtle) [reply data/mods/ssb/moves.js]: gdi I keep doing that.

File: data/mods/ssb/moves.js [hunk 8]
```diff
@@ -3305,34 +3305,73 @@ let BattleMovedex = {
 		target: "normal",
 		type: "Fire",
 	},
-	// Teclis
-	zekken: {
+// Teclis
```
Comment (ghost) [inline data/mods/ssb/moves.js]:
```suggestion
  // Teclis
```

**Comment by HoeenCoder** [comment]
You'll need to rebase this.

**Pull request closed without merging**